modules/common/topology/topology.go

```diff
@@ -52,12 +55,33 @@ func EnsureTopologyRef(
 	if err != nil {
 		return topology, hash, err
 	}
+
 	// Add finalizer (if not present) to the resource consumed by the Service
 	if controllerutil.AddFinalizer(topology, fmt.Sprintf("%s-%s", h.GetFinalizer(), finalizer)) {
 		if err := h.GetClient().Update(ctx, topology); err != nil {
 			return topology, hash, err
 		}
 	}
+
+	// Set default LabelSelector on topologyConstraints if not set, similar to cluster level default:
+	// https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/#cluster-level-default-constraints
+	topology = topology.DeepCopy()
+
+	topologyConstraints := topology.Spec.TopologySpreadConstraints
+	if topologyConstraints != nil {
+		for i := 0; i < len(*topologyConstraints); i++ {
+			current := &(*topologyConstraints)[i]
+			if current.LabelSelector == nil {
+				current.LabelSelector = defaultLabelSelector
```

Comment:
if `defaultLabelSelector` is nil (because I pass nil to this function) L75 will segfault the operator calling this function.
Similarly to what we do with `topologyConstraints` on L71, we should check in advance (likely L65) if `defaultLabelSelector` is nil, and eventually skip this block.

Reply:
operator need to handle nil LabelSelector either way

Reply:
I still think, if that parameter is `nil`, that it might be more efficient to not iterate on a `topologyConstraint` list and check every `.LabelSelector`, but hopefully we do not expect big lists, which may downgrade my previous comment to a nit if you tested and everything looks ok.

Reply:
After landing https://github.com/openstack-k8s-operators/glance-operator/pull/670 we can rebase `glance-operator` to incorporate this change

Reply:
I tested this locally (simply passing `nil` to the function to meet the signature) and I don't see big problems other the inefficient iteration that assigns `nil` to an already `nil` value in case a `defaultLabelSelector` is not provided.
I'll do a couple of more tests tomorrow with fresh eyes, but I'm planning to land this tomorrow if you think that we don't have to handle that case and skip the iteration.